sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFEvaluators.scala

```diff
@@ -131,6 +131,13 @@ class HiveGenericUDFEvaluator(
   def setArg(index: Int, arg: Any): Unit =
     deferredObjects(index).asInstanceOf[DeferredObjectAdapter].set(arg)
 
+  def setFuncArg(index: Int, arg: () => Any): Unit =
```

Comment:
It seems that `setFuncArg` and `setException` can be merged?

Reply:
It is a little ugly that we use inner class in codegen. If we use `setFuncArg`, the codegen for exceptions would be follow code.
```
        s"""
           |try {
           |  ${eval.code}
           |  if (${eval.isNull}) {
           |    $refEvaluator.setArg($i, null);
           |  } else {
           |    $refEvaluator.setArg($i, ${eval.value});
           |  }
           |} catch (Exception exp) {
           |  final exception = exp;
           |  $refEvaluator.setFundArg($i, new scala.Function0<Object>() {
           |    public Object apply() {
           |      throw exception;
           |    }
           |  });
           |}
           |""".stripMargin
```